donut/modules/marketplace/helpers.py

```diff
@@ -0,0 +1,754 @@
+import flask
+import sqlalchemy
+import re
+
+from donut.modules.core.helpers import get_member_data
+
+import routes
+
+# taken from donut-legacy, which was apparently taken from a CS11
+# C++ assignment by dkong
```

Comment:
Yeah, I got it from here:
http://users.cms.caltech.edu/~donnie/cs11/cpp/lab7/index.html

It looks like Donnie's completely redone the C++ track since